mcs/class/corlib/System/TermInfoDriver.cs

```diff
@@ -265,6 +264,8 @@ public void Init ()
 					cursorLeft = 0;
 					cursorTop = 0;
 				}
+
+				inited = true;
```

Comment:
I wonder if this needs to be done in a `try`/`finally` block - otherwise if initialization fails on the first thread, other threads will keep trying.  Not sure whether that's desirable.

/cc @steveisok and team

Reply:
I would prefer `try/finally` over this.

Reply:
@steveisok I've added `try/finally` as suggested.  Thanks for the feedback.